### PR TITLE
MAP-635 SAR endpoint

### PR DESCRIPTION
**Pull request by danbenton-mojdt**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsuofdataapi/config/OpenApiConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  @Bean
+  fun customOpenAPI(): OpenAPI = OpenAPI()
+    .servers(
+      listOf(
+        Server().url("/").description("Current url"),
+      ),
+    )
+    .info(
+      Info().title("Locations Inside Prison API")
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsuofdataapi/config/OpenApiConfiguration.kt:38]: Wording (wrong api)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsuofdataapi/config/JwtAuthHelper.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  internal fun createJwt(
+    subject: String?,
+    scope: List<String>? = listOf(),
+    roles: List<String>? = listOf(),
+    expiryTime: Duration = Duration.ofHours(1),
+    jwtId: String = UUID.randomUUID().toString(),
+  ): String =
+    mutableMapOf<String, Any>()
+      .also { subject?.let { subject -> it["user_name"] = subject } }
+      .also { it["client_id"] = "locations-inside-prison-client" }
```
Comment (Mjwillis) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsuofdataapi/config/JwtAuthHelper.kt:52]: client id

File: src/test/resources/application-test.yml [hunk 3]
```diff
@@ -4,3 +4,20 @@ server:
...
   info.cache.time-to-live: 0
+
+api:
+  base:
+    url:
+      oauth: http://localhost:8090/auth
+
+spring:
+  datasource:
+    url: jdbc:postgresql://localhost:9432/locations_inside_prison?sslmode=prefer
```
Comment (Mjwillis) [inline src/test/resources/application-test.yml:15]: wrong db

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsuofdataapi/utils/UserContextFilter.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package uk.gov.justice.digital.hmpps.hmppsuofdataapi.utils
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsuofdataapi/utils/UserContextFilter.kt:1]: Not sure we need these as we are not calling another API

**Review by Mjwillis — COMMENTED** [review]
Some comments

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsuofdataapi/config/OpenApiConfiguration.kt (hunk 1 above)
Comment (danbenton-mojdt) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsuofdataapi/config/OpenApiConfiguration.kt]: Fixed

File: src/test/resources/application-test.yml (hunk 3 above)
Comment (danbenton-mojdt) [reply src/test/resources/application-test.yml]: Fixed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsuofdataapi/utils/UserContextFilter.kt (hunk 4 above)
Comment (danbenton-mojdt) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsuofdataapi/utils/UserContextFilter.kt]: Removed

**Review by Mjwillis — APPROVED** [review]
Lgtm

**Pull request merged**